### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- [x] Make sure that you used the correct Gitflow.
- [x] Make sure that you document your work in a professional way.
### Project requirements
- [x] Interface
  - Create a class `Nameable`.
     - Implement a method called `correct_name` that will raise a `NotImplementedError`.
- [x] Turn your Person class to Nameable
  - Make sure that your Person class inherits from Nameable
  - Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.
- [x] Prepare base Decorator
  - Make sure that it inherits from Nameable.
  - In the constructor assign a nameable object from params to an instance variable.
  - Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.
- [x] Prepare CapitalizeDecorator and TrimmerDecorator
  - For the CapitalizeDecorator:
      - Create a class that inherits from the base Decorator class.
      - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
  - For the TrimmerDecorator:
      - Create a class that inherits from the base Decorator class.
      - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
```
Comment (julie-ify) [inline decorator.rb:3]: - [ ] You have created the Decorator and Trimmer classes, but you are yet to create the CapitalizeDecorator class, which should inherit from this class, for capitalizing the name. Kindly create it.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Benawi,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
##  Changes ♻️
- [x] Create the CapitalizeDecorator class

**Review by souad988 — APPROVED** [review]
Hi @Benawi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

approved2


## Highlights
Good communication with reviewer
Implement all requirements 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**